vector/src/main/java/im/vector/riotx/features/settings/devices/VectorSettingsDevicesFragment.kt

```diff
@@ -64,35 +62,19 @@ class VectorSettingsDevicesFragment @Inject constructor(
         recyclerView.configureWith(devicesController, showDivider = true)
         viewModel.observeViewEvents {
             when (it) {
-                is DevicesViewEvents.Loading -> showLoading(it.message)
-                is DevicesViewEvents.Failure -> showFailure(it.throwable)
-            }.exhaustive
-        }
-        viewModel.requestPasswordLiveData.observeEvent(this) {
-            maybeShowDeleteDeviceWithPasswordDialog()
-        }
-
-        viewModel.fragmentActionLiveData.observeEvent(this) { async ->
-            when (async) {
-                is Success -> {
-                    when (val action = async.invoke()) {
-                        is DevicesAction.PromptRename   -> {
-                            action.deviceInfo?.let { deviceInfo ->
-                                displayDeviceRenameDialog(deviceInfo)
-                            }
-                        }
-                        is DevicesAction.VerifyMyDevice -> {
-                            if (context is VectorBaseActivity) {
-                                VerificationBottomSheet.withArgs(
-                                        roomId = null,
-                                        otherUserId = action.userId!!,
-                                        transactionId = action.transactionId!!
-                                ).show(childFragmentManager, "REQPOP")
-                            }
-                        }
-                    }
+                is DevicesViewEvents.Loading            -> showLoading(it.message)
+                is DevicesViewEvents.Failure            -> showFailure(it.throwable)
+                is DevicesViewEvents.RequestPassword    -> maybeShowDeleteDeviceWithPasswordDialog()
+                is DevicesViewEvents.PromptRenameDevice -> displayDeviceRenameDialog(it.deviceInfo)
+                is DevicesViewEvents.ShowVerifyDevice   -> {
```

Comment:
For the view event, the device id is not needed, the transactionId is enough

Reply:
ok, thanks